base/util.go

```diff
@@ -666,4 +666,14 @@ func SingleHostCouchbaseURIToHttpURL(couchbaseUri string) (httpUrl string) {
 	// It made it past all checks.  Return a slice with a single string
 	return result.String()
 
+}
+
+// Slice a string to be less than or equal to desiredSze
+func SafeSlice(s string, desiredSize int) string {
+	if len(s) <= desiredSize {
+		return s
+	}
+
+	return s[:desiredSize]
+
```

Comment:
I think we've already got a SafeSlice in an in-flight PR from Andy, and it's a more generic slice function.

Can we rename this one 'StringPrefix' or something like that?